### ISSUE TITLE
Using JSX without importing React does not raise a warning/error

### ISSUE THREAD

**Issue by ide** [yannickcr/eslint-plugin-react#84]
Here is a sample JS file that demonstrates the issue. I define a couple of routes using JSX but don't explicitly reference React:

```js
// import React from 'react';
import { Route } from 'react-router';

import Home from './Home';
import Site from './Site';

let routes =
  <Route path="/" handler={Site}>
    <Route name="home" path="/" handler={Home} />
  </Route>;

export default routes;
```

This code fails to run because the JSX transformer (I use Babel) adds in `React.createElement(Route, ...)`, and React was not imported. The linter should complain that the JSX runtime is not defined. Also, when I uncomment `import React from 'react';` the linter complains that React is unused even though it is needed.

**Comment by yannickcr** [comment]
There is 2 rules to handle these case:
- [react-in-jsx-scope](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/react-in-jsx-scope.md): that will ensure React is available in the scope where JSX is used.
- [jsx-uses-react](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-uses-react.md): that will prevent the React variable to be incorrectly marked as unused.

Does theses 2 rules solve your problem ?

**Comment by ide** [comment]
Yes, that's perfect. Thank you for the pointers @yannickcr!
